data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/RemotePeerForwarder.java

```diff
@@ -231,23 +226,48 @@ private List<Record<Event>> populateBatchingQueue(final String destinationIp, fi
         return recordsFailedToBatch;
     }
 
-    private List<Record<Event>> forwardRecordBatches(final String destinationIp) {
-        final List<Record<Event>> recordsFailedToForward = new ArrayList<>();
+    private void forwardBatchedRecords() {
+        final Map<CompletableFuture<AggregatedHttpResponse>, List<Record<Event>>> futuresMap = peerBatchingQueueMap.keySet().stream()
+                .map(this::forwardRecordsForIp)
+                .flatMap(map -> map.entrySet().stream())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
 
-        List<Record<Event>> recordsToForward;
-        do {
-            recordsToForward = getRecordsToForward(destinationIp);
+        final CompletableFuture<Void> compositeFuture = CompletableFuture.allOf(futuresMap.keySet().toArray(CompletableFuture[]::new));
+        try {
+            compositeFuture.get();
+        } catch (final ExecutionException | InterruptedException e) {
+            LOG.debug("Caught exception getting results of composite forwarding future.", e);
```

Comment:
Is debug the right level for this?

Reply:
I used debug since the individual futures are processed below. I now see that I missed a log statement in the catch block of that though. Will add that quick